commcare_connect/utils/db.py

```
@@ -26,3 +30,16 @@ def slugify_uniquely(value, model, slugfield="slug"):
+
+def get_object_by_uuid_or_int(queryset, lookup_value, uuid_field):
+    lookup_str = str(lookup_value)
+
+    if lookup_str.isdigit():
+        return get_object_or_404(queryset, pk=int(lookup_str))
+
+    try:
+        uuid_val = uuid.UUID(lookup_str)
+        return get_object_or_404(queryset, **{uuid_field: uuid_val})
+    except ValueError:
+        raise Http404(f"No {queryset.model._meta.object_name} matches the given query.")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)